### PR TITLE
Added basic registry logs functionality

### PR DESCRIPTION
**Pull request by Muhammed-Moinuddin**
**Notes for Reviewers**
Just fetched and displayed registry logs via events. 
Kindly ignore UI part for now.

Screenshot from 2024-07-11 07-16-57

This PR fixes # 11208

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [X] Yes, I signed my commits.

### CODE REVIEW

**Comment by Muhammed-Moinuddin** [comment]
Kindly ignore that \n as well :D

File: ui/components/MesheryRegistrationLogs.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+const Loading = () => {
+  return (
+    <Box sx={{ display: 'flex', justifyContent: 'center' }}>
+      <CircularProgress />
+    </Box>
+  );
+};
+
+const EventsView = () => {
+  const events = useSelector(selectEvents) || [];
```
Comment (aabidsofi19) [inline ui/components/MesheryRegistrationLogs.js:17]: this is  a paginated filtered view of events .  directly make a rtk api call to events endpoint with filters applied to get the log events at take the latest

File: ui/components/MesheryRegistrationLogs.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+  const logsUrl = events.find((event) => event.category === 'registration')?.metadata?.ViewLink;
+  const [data, setData] = useState('No Logs Available');
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState(null);
+
+  useEffect(() => {
+    const fetchData = async () => {
```
Comment (aabidsofi19) [inline ui/components/MesheryRegistrationLogs.js:24]: use rtk api here

**Review by aabidsofi19 — CHANGES_REQUESTED** [review]

File: ui/components/MesheryRegistrationLogs.js (hunk 1 above)
Comment (Muhammed-Moinuddin) [reply ui/components/MesheryRegistrationLogs.js:17]: We don't have direct end-point for the logs via event. Do we need to create a separate rtk-query?

After hitting : `/api/v2/events`
Getting below:
`{"events":[{"acted_upon":"00000000-0000-0000-0000-000000000000","action":"get_summary","category":"entity","created_at":"2024-07-12T21:15:33.677091308+05:00","description":"For registrant successfully imported 9 models 520 relationships","id":"d25da155-f379-4ae1-a3b7-b8139efafc68","metadata":{"Hostname":""},"operation_id":"bc78cf3a-4ded-487d-a6c5-ac47ac57a5e3","severity":"informational","status":"unread","system_id":"ed13ec4d-c19a-451f-9faf-1819c17a8f08","updated_at":"2024-07-12T21:15:33.677186219+05:00","user_id":"ed13ec4d-c19a-451f-9faf-1819c17a8f08"},{"acted_upon":"00000000-0000-0000-0000-000000000000","action":"get_summary","category":"entity","created_at":"2024-07-12T21:15:33.677716596+05:00","description":"For registrant Artifacthub successfully imported 603 models 27605 components","id":"3f143fd1-b5d9-43fe-9bea-7e93f8e92ac0","metadata":{"Hostname":"Artifacthub"},"operation_id":"258fc289-69a6-43b3-a69e-2281b1779bc2","severity":"informational","status":"unread","system_id":"ed13ec4d-c19a-451f-9faf-1819c17a8f08","updated_at":"2024-07-12T21:15:33.677797557+05:00","user_id":"ed13ec4d-c19a-451f-9faf-1819c17a8f08"},{"acted_upon":"00000000-0000-0000-0000-000000000000","action":"get_summary","category":"registration","created_at":"2024-07-12T21:15:33.677980533+05:00","description":"Failed to import 18 components","id":"9fe6d3d4-6312-45bc-a46a-d79a15ebe91c","metadata":{"DownloadLink":"/home/moin/.meshery/logs/registry/registry-logs.log","LongDescription":"The import process for a registrant Artifacthub encountered difficulties,due to which Failed to import 18 components. Specific issues during the import process resulted in certain entities not being successfully registered in the table.","SuggestedRemediation":"Visit docs with the error code https://docs.meshery.io/reference/error-codes","ViewLink":"/home/moin/.meshery/logs/registry/registry-logs.log"},"operation_id":"e29d9175-8800-46b9-9b09-7e108ac32880","severity":"error","status":"unread","system_id":"ed13ec4d-c19a-451f-9faf-1819c17a8f08","updated_at":"2024-07-12T21:15:33.678019699+05:00","user_id":"ed13ec4d-c19a-451f-9faf-1819c17a8f08"},{"acted_upon":"00000000-0000-0000-0000-000000000000","action":"get_summary","category":"entity","created_at":"2024-07-12T21:15:33.678178134+05:00","description":"For registrant Github successfully imported 50 models 6005 components","id":"fa57ab57-566f-49a2-aebc-bf37a4ce4923","metadata":{"Hostname":"Github"},"operation_id":"7895c3c7-d9d8-4aaa-9b2a-31e2b80432ac","severity":"informational","status":"unread","system_id":"ed13ec4d-c19a-451f-9faf-1819c17a8f08","updated_at":"2024-07-12T21:15:33.678224375+05:00","user_id":"ed13ec4d-c19a-451f-9faf-1819c17a8f08"}],"page":0,"page_size":25,"count_by_severity_level":[],"total_count":4}`

Should I fetch 2 times? One for reaching events via `/api/v2/events` and getting the required url (attached with ViewLink), then hitting the target url i.e. /home/moin/.meshery/logs/registry/registry-logs.log to get the logs data?

Are you talking about directly hitting the below api?
https://playground.meshery.io/api/system/fileView?file=%2Fhome%2Fappuser%2F.meshery%2Flogs%2Fregistry%2Fregistry-logs.lo

Pretty confused, kindly elaborate a bit.

Comment (Muhammed-Moinuddin) [reply ui/components/MesheryRegistrationLogs.js:17]: @aabidsofi19

**Pull request closed without merging**